invenio_app/ext.py

```diff
@@ -64,6 +65,13 @@ def ping():
 
             app.register_blueprint(blueprint)
 
+        @app.before_request
+        def set_request_id():
+            """Extracts X-Request-Id and sets in the flask variable."""
+            key = current_app.config.get('APP_REQUEST_ID_HEADER_NAME')
+            r_id = request.headers.get(key)
+            request.request_id = r_id
```

Comment:
I think this should be using the [Flask g object](http://flask.pocoo.org/docs/0.12/api/#flask.g) instead.

```python
g.request_id = request.headers.get(
    current_app.config.get('APP_REQUEST_ID_HEADER'))
```